specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,318 @@
+# Background
+
+We are exposing an event that will be raised when an attempt to launch a protocol that is registered with the OS (external protocol) is made.
+When navigating to a URI, the URI scheme determines how to handle the URI.
+Some schemes like http, and https, are resolved by WebView2 and the navigation is handled in the WebView2.
+Other URI schemes may be registered externally to the WebView2 with the OS by other applications.
+Such schemes are external protocols and are handled by launching the registered application with the URI.
+
+The host will be given the option to cancel the external protocol launch with the `LaunchingExternalProtocol` event.
+Cancelling the launch gives the host the opportunity to hide the default dialog, display a custom dialog, and then launch the external protocol themselves.
+
+# Description
+
+This event will be raised before the external protocol launch occurs.
+When an attempt to launch an external protocol is made, the default dialog is displayed in which the user can select `Open` or `Cancel` if the host does not cancel the event.  
+The `NavigationStarting` event will be raised before the `LaunchingExternalProtocol` event, followed by the `NavigationCompleted` event.

```

Comment:
What does NavigationCompleted report for external protocol launches?

| Scenario | IsSuccess | HttpStatusCode | WebErrorStatus | WebView2.Source |
|-|-|-|-|-|
| User opens | True? | 200? | N/A | Unchanged |
| User cancels | False? | ? | ConnectionAborted? | Unchanged | 
| App cancels | ? | ? | ? | Unchanged |

From the fact that SourceChanged is not raised, this implies that WebView2.Source is unchanged. Is that true?

If we return IsSuccess=true for a successful external protocol launch, could lead to a weird state where a successful navigation does *not* change the WebView2.Source to the navigated-to URI.

Reply:
We report a NavigationCompleted event with IsSuccess = false and WebErrorStatus = ConnectionAborted regardless of whether or not the external protocol launches. Some existing apps rely on the Navigation events during this scenario so we decided to maintain the existing Navigation events for external protocol launches.
The WebView2.Source remains unchanged.

Reply:
Please ensure we document that for the event: IsSuccess = false and WebErrorStatus = ConnectionAborted, and Source is unchanged regardless of what happens in the event handler.